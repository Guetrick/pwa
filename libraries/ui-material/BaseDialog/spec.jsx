import React from 'react';
import { shallow } from 'enzyme';
import Button from '@shopgate/pwa-ui-shared/Button';
import BasicDialog from './index';

describe('<BasicDialog />', () => {
  const props = {
    actions: [
      {
        label: 'action0',
        action: jest.fn(),
      },
      {
        label: 'action1',
        action: jest.fn(),
      },
      {
        label: 'action2',
        action: jest.fn(),
      },
    ],
  };

  it('should render with minimal props', () => {
    const wrapper = shallow(<BasicDialog actions={[]} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should show correct number of buttons', () => {
    const wrapper = shallow(<BasicDialog {...props} />);

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find(Button).length).toBe(props.actions.length);
  });

  it('should trigger correct actions', () => {
    const wrapper = shallow(<BasicDialog {...props} />);

    const button = wrapper.find(Button).at(1);
    button.simulate('click');

    expect(wrapper).toMatchSnapshot();
    expect(props.actions[0].action).not.toBeCalled();
    expect(props.actions[1].action).toBeCalled();
    expect(props.actions[2].action).not.toBeCalled();
  });

  it('should render with node as a title', () => {
    const wrapper = shallow(<BasicDialog title={<strong>My Title</strong>} />);
    expect(wrapper.find('strong').text()).toEqual('My Title');
  });
});
