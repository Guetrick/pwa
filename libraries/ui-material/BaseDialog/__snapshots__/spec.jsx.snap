// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicDialog /> should render with minimal props 1`] = `
<div
  className="css-hrxvvd"
  data-test-id="basicDialog"
>
  <div
    className="css-1sirlct"
  />
  <div
    className="css-1h5vk0a"
  >
    <div
      className="css-hw75se"
    />
  </div>
</div>
`;

exports[`<BasicDialog /> should show correct number of buttons 1`] = `
<div
  className="css-hrxvvd"
  data-test-id="basicDialog"
>
  <div
    className="css-1sirlct"
  />
  <div
    className="css-1h5vk0a"
  >
    <div
      className="css-hw75se"
    >
      <Button
        className="css-7vankg"
        disabled={false}
        flat={true}
        key="action0"
        onClick={[MockFunction]}
        testId="Button"
        type="primary"
        wrapContent={true}
      >
        <Translate
          className=""
          params={Object {}}
          role={null}
          string="action0"
          transform={null}
        />
      </Button>
      <Button
        className="css-7vankg"
        disabled={false}
        flat={true}
        key="action1"
        onClick={[MockFunction]}
        testId="Button"
        type="primary"
        wrapContent={true}
      >
        <Translate
          className=""
          params={Object {}}
          role={null}
          string="action1"
          transform={null}
        />
      </Button>
      <Button
        className="css-7vankg"
        disabled={false}
        flat={true}
        key="action2"
        onClick={[MockFunction]}
        testId="Button"
        type="primary"
        wrapContent={true}
      >
        <Translate
          className=""
          params={Object {}}
          role={null}
          string="action2"
          transform={null}
        />
      </Button>
    </div>
  </div>
</div>
`;

exports[`<BasicDialog /> should trigger correct actions 1`] = `
<div
  className="css-hrxvvd"
  data-test-id="basicDialog"
>
  <div
    className="css-1sirlct"
  />
  <div
    className="css-1h5vk0a"
  >
    <div
      className="css-hw75se"
    >
      <Button
        className="css-7vankg"
        disabled={false}
        flat={true}
        key="action0"
        onClick={[MockFunction]}
        testId="Button"
        type="primary"
        wrapContent={true}
      >
        <Translate
          className=""
          params={Object {}}
          role={null}
          string="action0"
          transform={null}
        />
      </Button>
      <Button
        className="css-7vankg"
        disabled={false}
        flat={true}
        key="action1"
        onClick={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        testId="Button"
        type="primary"
        wrapContent={true}
      >
        <Translate
          className=""
          params={Object {}}
          role={null}
          string="action1"
          transform={null}
        />
      </Button>
      <Button
        className="css-7vankg"
        disabled={false}
        flat={true}
        key="action2"
        onClick={[MockFunction]}
        testId="Button"
        type="primary"
        wrapContent={true}
      >
        <Translate
          className=""
          params={Object {}}
          role={null}
          string="action2"
          transform={null}
        />
      </Button>
    </div>
  </div>
</div>
`;
